File: javascript/fizzbuzz.js
var generate = function(min, max){
  var fb = [], arr = [], i;
  min = min == void null ? 1 : min;
  max = max == void null ? 100 : max;
  for(i = min; i <= max; i++){
    fb = say(i);
    arr.push(fb.length === 0 ? i : fb);
  }
  return arr;
};

var say = function(x){
  var fb = [];
  if(x % 3 === 0){
    fb.push('Fizz');
  }
  if(x % 5 === 0){
    fb.push('Buzz');
  }
  return fb.join('');
};

module.exports = {
  generate: generate
};

File: c/fizzbuzz.c
#include "fizzbuzz.h"

void say(char* str, int x){
  if(x % 15 == 0)
    strcpy(str, "FizzBuzz");
  else if(x % 3 == 0)
    strcpy(str, "Fizz");
  else if(x % 5 == 0)
    strcpy(str, "Buzz");
  else
    sprintf(str, "%d", x);
};

char **generateNoParams(){
  generate(1, 100);
}

char **generate(int min, int max){
  int i = min;
  int j = 0;
  char ** sub_str = malloc(max * sizeof(char*));
  for (; i <= max; i++, j++){
    sub_str[j] = malloc(9 * sizeof(char));
    say(sub_str[j], i);
  }
  
  return sub_str;
};